asv/commands/run.py

```diff
@@ -386,11 +391,15 @@ def iter_rounds_commits():
                         params['python'] = env.python
                         params.update(env.requirements)
 
-                        skip_save = (dry_run or isinstance(env, environment.ExistingEnvironment))
+                        skip_save = dry_run or (isinstance(env, environment.ExistingEnvironment)
+                                                and commit_label is None)
 
                         skip_list = skipped_benchmarks[(commit_hash, env.name)]
                         benchmark_set = benchmarks.filter_out(skip_list)
 
+                        if isinstance(env, environment.ExistingEnvironment) and commit_label is not None:
+                            commit_hash = 'label-' + commit_label.replace('-', '_')
+
```

Comment:
I replace the `'-'` characters by `'_'` because the result filename is split by `'-'` in several places. It should avoid weird behaviors, but may modify the label provided by a user.